### ISSUE TITLE
Inconsistent NULL behavior

### ISSUE THREAD

**Issue by zmitchell** [antoyo/tql#21]
Say that you create a model like this:
```rust
#[derive(SqlTable)]
struct MyModel {
    id: PrimaryKey,
    foo: Option<String>,
}
```
Now try to insert a value that has `None` in the `foo` field.
```rust
let id = sql!(MyModel.insert(foo = None));
```
You'll get a type error saying that `foo` should be of type `String`. If you want to disallow `NULL` fields, you should also disallow creating a model containing `Option<T>`.

**Comment by antoyo** [comment]
Oh, actually, you just need to avoid specifiying the value, but I guess this might not always be what we want, i.e. when we have a variable of type `Option<String>`.
So, the following code works:
```rust
let id = sql!(MyModel.insert());
```
So, I guess accepting both the types `String` and `Option<String>` would fix this issue.
Thanks for reporting this issue.

**Comment by zmitchell** [comment]
I'm not sure I follow. What do you mean "you just need to avoid specifying the value"? It also fails when you try to insert a record with more than one field (with at least one of them being `Option<T>`).

**Comment by antoyo** [comment]
The current way to set a value to `NULL` is to not specify it in the `insert()` method call:
```rust
let id = sql!(MyModel.insert(/* the value is not here*/);
```
If you want to insert a value instead of `NULL`, specify it without `Some`.
```rust
let id = sql!(MyModel.insert(foo = "string");
```
Just try the code I posted previously using the model from your first post and it should work.
Can you show me an example for your second point?
Thank you.

**Comment by antoyo** [comment]
Fixed by #23 and tested in #27.
